core/src/main/java/io/spine/core/Events.java

```diff
@@ -183,16 +200,80 @@ public static EventId checkValid(EventId id) {
 
     /**
      * Obtains the {@link TenantId} from the given {@link Event}.
+     * 
+     * <p>The {@code TenantId} is retrieved by traversing the passed {@code Event}s context. It is
+     * stored in the initial {@link CommandContext} and can be retrieved from the events origin 
+     * command or rejection context. 
      */
     @Internal
     public static TenantId getTenantId(Event event) {
-        final TenantId result = event.getContext()
-                                     .getCommandContext()
-                                     .getActorContext()
-                                     .getTenantId();
+        checkNotNull(event);
+        
+        final Optional<CommandContext> commandContext = getCommandContext(event);
+
+        if (!commandContext.isPresent()) {
+            return TenantId.getDefaultInstance();
+        }
+
+        final TenantId result = getTenantId(commandContext.get());
         return result;
     }
 
+    /**
+     * Obtains the {@link TenantId} from the {@link CommandContext}.
+     * 
+     * <p>The {@link CommandContext} is accessible from the {@link Event} if the {@code Event} was 
+     * created as a result of some command or its rejection. This makes the {@code CommandContext} 
+     * a valid {@code TenantId} source inside of the {@code Event}.
+     */
+    private static TenantId getTenantId(CommandContext context) {
```

Comment:
I would put this into the `Commands` class. We have already `getTenantId(Command)` there. 

This way we localize the knowledge about whereabouts `TenantId` is in a command “guts” inside one utility class `Commands`.